src/content/docs/en/reference/api-reference.mdx

```diff
@@ -619,6 +619,14 @@ This property is only available when building for SSR (server-side rendering) an
 
 The locale computed from the current URL, using the syntax specified in your `locales` configuration. If the URL does not contain a `/[locale]/` prefix, then the value will default to `i18n.defaultLocale`.
 
+### `Astro.routePattern`
+
+The current route pattern being rendered. The pattern will look like a file path, without the [`srcDir`](/en/reference/configuration-reference/#srcdir) and the `pages` folder. The path will start with a leading slash, and it won't have the file extension.
```

Comment:
Is a "route pattern" a known thing, like that there are docs defining it at MDN etc., or is this something we have defined? This first sentence feels like a bit of a circular definition: "routePattern is a route pattern" and that is supposed to be meaningful. :smile: It would be nice if we could link to an external source for "route pattern" if that is a standard concept. And if it's not, then we have to think of a way to describe it that is meaningful.

I'm also not clear when this would be used. Can we get an example showing actual usage? When would someone need this, and how would they use this information if they had it? Would you use this in a regular static Astro site? Is this more expected to be used by integration authors? 

Is it about build time knowledge of which file is currently being processed to create a route, or about knowing which component created the page a visitor is currently visiting? Does it require SSR?

There are lots of questions to answer here, and I think focusing on what this is used for will help!

(Also reminder we'll need a `<Since />` component, too.